ktor-server/jvm/src/io/ktor/server/application/plugins/api/ServerPlugin.kt

```diff
@@ -274,12 +326,68 @@ public abstract class ServerPlugin<Configuration : Any> private constructor(
          *          println(call.request.uri)
          *      }
          * }
+         *
+         * application.install(MyPlugin)
          * ```
          **/
-        public fun createPlugin(
+        public fun createApplicationPlugin(
             name: String,
             body: ServerPlugin<Unit>.() -> Unit
-        ): ServerPluginFactory<Unit> = createPlugin<Unit>(name, {}, body)
+        ): ServerApplicationPluginFactory<Unit> = createApplicationPlugin(name, {}, body)
+
+        /**
+         * Creates a [ServerPluginFactory] that can be installed into [Application].
+         * A canonical way to create a [ServerPlugin] without any configuration.
+         *
+         * @param name A name of your new plugin that will be used if you need find an instance of
+         * your plugin when it is installed to an [Application].
+         * @param body Allows you to define handlers ([onCall], [onCallReceive], [onCallRespond] and so on) that
+         * can modify the behaviour of an [Application] where your plugin is installed.
+         *
+         * Usage example:
+         * ```
+         * val MyPlugin = createPlugin("MyPlugin") {
+         *      // This block will be executed when you call install(MyPlugin)
+         *
+         *      onCall { call ->
+         *          // Prints requested URL each time your application receives a call:
+         *          println(call.request.uri)
+         *      }
+         * }
+         *
+         * route("a") {
+         *   install(MyPlugin)
+         * }
+         * ```
+         **/
+        public fun createRoutingScopedPlugin(
+            name: String,
+            body: RoutingScopedPlugin<Unit>.() -> Unit
+        ): ServerRoutingScopedPluginFactory<Unit> = createRoutingScopedPlugin(name, {}, body)
+
+        private fun <Configuration : Any, Plugin : ServerPlugin<Configuration>> Plugin.setupPlugin(body: Plugin.() -> Unit) {
+            apply(body)
+
+            pipelineHandlers.forEach { handle ->
+                handle(pipeline)
+            }
+
+            callInterceptions.forEach {
+                it.action(pipeline)
+            }
+
+            onReceiveInterceptions.forEach {
+                it.action(pipeline.receivePipeline)
+            }
+
+            onResponseInterceptions.forEach {
+                it.action(pipeline.sendPipeline)
+            }
+
+            afterResponseInterceptions.forEach {
+                it.action(pipeline.sendPipeline)
+            }
+        }
     }
 
```

Comment:
`createRoutingScopedPlugin` function with `Configuration` is missing. Is there a reason for that?

Reply:
It is here in this file (https://github.com/ktorio/ktor/pull/2306/files/9dd5d45bafef0d87ad0714c48fce4360dcbfc00e#diff-244208d28b7b8aa486f55d16a600eae10295b68b68029cf9c54d14af37d4ace5R286)
Or am I missing something?

Reply:
Your link says, "We went looking everywhere, but couldn’t find those commits."